### DISCUSSION TITLE
Personal Access Token (PAT) for Users

### DISCUSSION THREAD

**Discussion by danielmichalichyn** [zitadel/zitadel#6503] [Q&A] [answered]
Hello guys, 

I have a scenario where I have a project with a registered API application. To give access to this API i've use service users with tokens created with specific roles. However, in these same organizations, I have registered users through an Identity Provider, it is possible for these users to generate an Access Tokens, or something similar, so that I can give access to my API. What is the best approach here?
I've hope that i made myself clear, if you have any doubt about my current setup, feel free to ask and i'll try to clarify this a little better :) 

Thanks in advance.

**Comment by mffap** [comment]
Hi @danielmichalichyn I'm guessing, but perhaps you are not passing the scope `urn:zitadel:iam:org:project:id:<API_PROJECT_ID>:aud` (replace the variable with your api's project id) when authenticating your users. Can you try adding this scope and see if users can use the access token on the API endpoint (eg, validate the token on the introspect).

We have an example guide and application for this scenario: https://zitadel.com/docs/guides/solution-scenarios/frontend-calling-backend-API

**Reply by kbumsik** [reply]
Getting an normal access token sometimes doesn't work for some business cases because of it's sort lifetime.

Many API providers, such as OpenAPI, allows users to create non-expiring API Keys, or configurable expiration. And it is exactly what Service Users' PAT does. Maybe this is what the OP is thinking?

**Reply by danielmichalichyn** [reply]
Actually i have a scenario a little bit different. I do have a Web App and the login is provided by Zitadel, and i've also have a standalone API that i've registered inside the same project in my Zitadel. My main question here is if this user created to log into my Web application can also have a token (PAT) to access my API? I've understood your suggestion @mffap , but in that case the API would need to be dependent on my Web Application, and this is not the way i was intended to go. I don't know if you have any other suggestions, or maybe my way of thinking here is wrong. I do think that my objective here is somewhat what @kbumsik  was suggesting but with the user account having the token and not the service account.

**Comment by hifabienne** [comment, marked as answer]
At the moment we do not have PAT for "human" users.
We are currently talking about the possibility to have User Schemas, which should also allow to set which authentication methods are possible on that user schema.
At the moment the implementation of PAT for human users is completaly separat to the one of the service accounts, and we do not want to duplicate that logic. So we will wait till the User Schema Problem is solved to have a look at PAT for human users.
Issue: #6433

**Reply by danielmichalichyn** [reply]
hmm ok, thanks for the answer @hifabienne so i'll be waiting for this to be implemented :)

**Reply by roquie** [reply]
We are also waiting for implementation, really necessary functionality

**Reply by danielloader** [reply]
Figured it's worth asking a year on with the User Schema work under heavy and active development if this functionality is still planned?

**Reply by hifabienne** [reply]
hi @danielloader 
its still planned, we actually had other prioritise, but started now with the implementation of the user schemas

**Reply by danielloader** [reply]
Any further update now?

**Reply by pmahindrakar-oss** [reply]
+1 any further update ?

**Reply by rajatcing** [reply]
hey @danielloader @pmahindrakar-oss we are not planning to bring PAT for users(human) at the moment
